src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs

```diff
@@ -86,13 +97,17 @@ private static void Run(this IWebHost host, CancellationToken token, string shut
                     Console.WriteLine(shutdownMessage);
                 }
 
-                token.Register(state =>
+                token.Register(() =>
+                {
+                    applicationLifetime.StopApplication();
```

Comment:
In the original code the `applicationLifetime` was passed as a parameter. I think that was done to avoid extra code generation for capturing local variables into the lambda.